src/stories/task-tracker.tsx

```diff
@@ -0,0 +1,78 @@
+import { useState, useRef } from "react";
+import { FaTrashCan } from "react-icons/fa6";
+
+export default function TaskTracker() {
+    const [tasks, setTasks] = useState<{ text: string; marked: boolean }[]>([]);
+    const [inputValue, setInputValue] = useState("");
+    const inputRef = useRef<HTMLInputElement>(null);
+
+    const addTask = () => {
+        setTasks(
+            [...tasks, { text: inputValue, marked: false }].sort(
+                (a, b) => Number(a.marked) - Number(b.marked)
+            )
+        );
+        setInputValue("");
+        inputRef.current?.focus();
+    };
+
+    const checkTask = (index: number) => {
+        const updatedTasks = tasks.slice();
+        updatedTasks[index].marked = !updatedTasks[index].marked;
+        setTasks(
+            updatedTasks.sort((a, b) => Number(a.marked) - Number(b.marked))
```

Comment:
Using `.sort()` here updates the actual data in the state. This is not optimal. It would be great to instead keep the order in the state as is and sort the tasks in the rendering. So just moving this sort function to line 53 (before the .map function call) should solve that.